### DISCUSSION TITLE
Resizing all widgets when window size changes

### DISCUSSION THREAD

**Discussion by ChanaGoldstein** [TomSchimansky/CustomTkinter#1932] [Q&A]
Hi,
I'm trying to figure out how to make all the widgets resize when I drag and drop the main window size using my mouse.
In the example provided here: https://github.com/TomSchimansky/CustomTkinter/blob/master/examples/complex_example.py the widgets inside the window change size when I resize the main window.
![image](https://github.com/TomSchimansky/CustomTkinter/assets/39883332/01a09d33-7c3b-4713-ae18-4c149ffb9116)

I'm trying to do something similar, and here is my code:
`
self.flashcard_window = ctk.CTkToplevel(self.window)

        self.flashcard_window.title(f"{self.current_deck_name} Flashcards")
        self.flashcard_window.protocol("WM_DELETE_WINDOW",
                                       self.on_closing_deck)
        self.flashcard_window.wm_transient(self.window)

        self.question_label = ctk.CTkLabel(self.flashcard_window,
                                           text='Question', wraplength=WARP_LENGTH,
                                           justify="center")
        self.question_label.grid(row=1, column=0, padx=(20, 20), pady=(20, 0))

        self.question_media_label = ctk.CTkLabel(self.flashcard_window,
                                                 image=None, text='')
        self.question_media_label.grid(row=2, column=0, padx=(20, 20),
                                       pady=(20, 0))

        self.answer_button = ctk.CTkButton(self.flashcard_window,
                                           text="Show Answer",
                                           command=self.show_answer)
        self.answer_button.grid(row=3, column=0, padx=(20, 20), pady=(20, 0))

        self.answer_label = ctk.CTkLabel(self.flashcard_window, text="",
                                         font=(self.font_name, self.font_size),
                                         wraplength=WARP_LENGTH,
                                         justify="center")
        self.answer_label.grid(row=4, column=0, padx=(20, 20), pady=(20, 0))

        self.answer_media_label = ctk.CTkLabel(self.flashcard_window,
                                               image=None, text='')
        self.answer_media_label.grid(row=5, column=0, padx=(20, 20),
                                     pady=(20, 0))

        self.rating_frame = ctk.CTkFrame(self.flashcard_window)
        self.rating_frame.grid(row=6, column=0, padx=(20, 20), pady=(20, 0))

        self.rating_label = ctk.CTkLabel(self.rating_frame,
                                         text="How well did you know the "
                                              "answer?")
        self.rating_label.grid()
        self.rating_value = tk.IntVar()
        self.rating_buttons = []

        for i in range(1, 6):
            rating_button = ctk.CTkRadioButton(self.rating_frame, text=str(i),
                                               variable=self.rating_value,
                                               value=i,
                                               command=self.next_flashcard)
            self.rating_buttons.append(rating_button)

            rating_button.grid(row=7, column=i, padx=(10, 10), pady=(20, 20))

        self.num_cards_left_label = ctk.CTkLabel(self.flashcard_window,
                                                 text="")
        self.num_cards_left_label.grid(row=9, column=0, padx=(20, 20),
                                       pady=(20, 20))

        self.delete_card_button = ctk.CTkButton(self.flashcard_window,
                                                text='Delete',
                                                command=self.on_delete_card)
        self.delete_card_button.grid(row=2, column=1, padx=(20, 20),
                                     pady=(20, 20))

        self.edit_card_button = ctk.CTkButton(self.flashcard_window,
                                              text='Edit',
                                              command=self.on_edit_card)
        self.edit_card_button.grid(row=3, column=1, padx=(20, 20),
                                   pady=(20, 20))
`
However, when I try to resize the window, the widgets get cut off:

Before resizing:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/39883332/ff53d25e-ff36-426e-abe8-808a17574816)

After resizing:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/39883332/2a190d7a-5b48-4701-bb7c-5c51706957fe)

I've looked at the following posts, but I couldn't figure out how to adapt the answers provided in them to my example.

https://github.com/TomSchimansky/CustomTkinter/discussions/435

https://github.com/TomSchimansky/CustomTkinter/wiki/App-structure-and-layout#small-onefile-example-app-with-grid (Here I understood the simple example but I couldn't figure out how to modify the grid configuration so it'll have the layout I want).

https://github.com/TomSchimansky/CustomTkinter/discussions/387

I would appreciate your help!

Thanks :)

**Comment by Akascape** [comment]
@ChanaGoldstein Study about the grid weight methods. (`grid_rowconfigure` and `grid_columnconfigure` methods, and also parameters like sticky)
